### PR TITLE
Adapt wire-server charts so they can be used to run wire cloud

### PR DESCRIPTION
**Pull request by akshaymankar**
charts/{brig,cargohol,galley,gundeck}: Allow not configuring AWS credentials and allow using a special service account.
This way, when operating wire in AWS cloud either instance profiles or IAM role attached to a service account can be used to communicate with AWS.

Allow new configurations in the brig chart:
* `config.emailSMS.user.invitationUrl`
* `config.emailSMS.team.tInvitationUrl`
* `config.emailSMS.team.tActivationUrl`
* `config.emailSMS.team.tCreatorWelcomeUrl`
* `config.emailSMS.team.tMemberWelcomeUrl`
* `config.setProviderSearchFilter`
* `config.setWhitelist`
* `config.setFeatureFlags`
* `config.setCustomerExtensions`

If any values in config.emailSMS.team are specified, all must be specified.

Allow new configurations in the gundeck chart:
* `config.perNativePushConcurrency`
* `config.maxConcurrentNativePushes.soft`
* `config.maxConcurrentNativePushes.hard`

Other changes:
* Default `maxTeamSize` changed from 500 to 10000. (larger teams have been supported by wire-server code for a while; but the default value had not been increased yet)

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [ ] If this PR changes development workflow or dependencies, they have been A) automated and B) documented under docs/developer/. All efforts have been taken to minimize development setup breakage or slowdown for co-workers.
 - [ ] If HTTP endpoint paths have been added or renamed, the **endpoint / config-flag checklist** (see Wire-employee only backend [wiki page](https://github.com/zinfra/backend-wiki/wiki/Checklists)) has been followed.
 - [ ] If a cassandra schema migration has been added, I ran **`make git-add-cassandra-schema`** to update the cassandra schema documentation.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.
   - [ ] If new config options introduced: added usage description under docs/reference/config-options.md
   - [ ] If new config options introduced: recommended measures to be taken by on-premise instance operators.
   - [ ] If a cassandra schema migration is backwards incompatible (see also [these docs](https://github.com/wireapp/wire-server/blob/develop/docs/developer/cassandra-interaction.md#cassandra-schema-migrations)), measures to be taken by on-premise instance operators are explained.
   - [ ] If a data migration (not schema migration) introduced: measures to be taken by on-premise instance operators.
   - [ ] If public end-points have been changed or added: does nginz need un upgrade?
   - [ ] If internal end-points have been added or changed: which services have to be deployed in a specific order?

### CODE REVIEW

**Review by jschaul — APPROVED** [review]
This iteration of helm chart changes is working, LGTM.

A changelog entry would be nice to add, though.

**Comment by jschaul** [comment]
CI seems to be failing though, still.

**Review by jschaul — APPROVED** [review]
Looks good, thanks for the detailed description!

File: changelog.d/2-features/no-aws-creds [hunk 1]
```diff
@@ -0,0 +1,2 @@
+charts/{brig,cargohol,galley,gundeck}: Allow not configuring AWS credentials and allow using a special service account.
```
Comment (flokli) [inline changelog.d/2-features/no-aws-creds:1]: `s/cargohol/cargohold`

small typo

Comment (akshaymankar) [reply changelog.d/2-features/no-aws-creds:1]: Fixed here: https://github.com/wireapp/wire-server/pull/2399

**Pull request merged**